id: PYSEC-0000-CVE-2021-29557
package:
  name: tensorflow
  ecosystem: PyPI
details: TensorFlow is an end-to-end open source platform for machine learning. An
  attacker can cause a denial of service via a FPE runtime error in `tf.raw_ops.SparseMatMul`.
  The division by 0 occurs deep in Eigen code because the `b` tensor is empty. The
  fix will be included in TensorFlow 2.5.0. We will also cherrypick this commit on
  TensorFlow 2.4.2, TensorFlow 2.3.3, TensorFlow 2.2.3 and TensorFlow 2.1.4, as these
  are also affected and still in supported range.
affects:
  ranges:
  - type: GIT
    repo: https://github.com/tensorflow/tensorflow
    fixed: 7f283ff806b2031f407db64c4d3edcda8fb9f9f5
  - type: ECOSYSTEM
    fixed: 2.1.4
  - type: ECOSYSTEM
    introduced: 2.2.0
    fixed: 2.2.3
  - type: ECOSYSTEM
    introduced: 2.3.0
    fixed: 2.3.3
  - type: ECOSYSTEM
    introduced: 2.4.0
    fixed: 2.4.2
references:
- type: WEB
  url: https://github.com/tensorflow/tensorflow/commit/7f283ff806b2031f407db64c4d3edcda8fb9f9f5
- type: WEB
  url: https://github.com/tensorflow/tensorflow/security/advisories/GHSA-xw93-v57j-fcgh
aliases:
- CVE-2021-29557
modified: "2021-05-18T19:59:00Z"
published: "2021-05-14T20:15:00Z"
